### PR TITLE
[TEX-537] Refactor compression

### PR DESCRIPTION
**Pull request by timdawborn**
This PR continues to cleanup this codebase ahead of supporting richer, content-type-aware, request body matching.

This PR refactors all of the logic that deals with compression into a singular location, and removes an old, outdated third-party library in favour of the builtin NodeJS implementation.

### CODE REVIEW

File: src/persistence.ts [hunk 1]
```diff
@@ -120,22 +122,12 @@ export function reviveTape(persistedRecord: PersistedTapeRecord): TapeRecord {
...
-  const originalBuffer = buffer;
-  let compression: CompressionAlgorithm = "none";
-  if (contentEncoding === "br") {
-    buffer = Buffer.from(brotli.decompress(buffer));
-    compression = "br";
-  }
-  if (contentEncoding === "gzip") {
-    buffer = gunzipSync(buffer);
-    compression = "gzip";
-  }
```
Comment (timdawborn) [inline src/persistence.ts:138]: Moved this common logic into `compression.ts`.

File: src/persistence.ts [hunk 2]
```diff
@@ -170,21 +163,7 @@ function unserialiseBuffer(persisted: PersistedBuffer): Buffer {
...
-          buffer = Buffer.from(compressed);
-        } else {
-          throw new Error(`Brotli compression failed!`);
-        }
-      }
-      if (persisted.compression === "gzip") {
-        // TODO: Find a workaround for the new compressed message not necessarily
-        // being identical to what was originally sent (update Content-Length?).
-        buffer = gzipSync(buffer);
-      }
```
Comment (timdawborn) [inline src/persistence.ts:187]: Moved this common logic into `compression.ts`.

File: src/http.ts [hunk 3]
```diff
@@ -49,35 +51,28 @@ export function getHeaderAsString(
...
-  );
-  switch (contentEncoding) {
-    case "":
-      return request.body;
-    case "br":
-      return Buffer.from(brotli.decompress(request.body));
-    case "gzip":
-      return zlib.gunzipSync(request.body);
-    default:
-      throw Error(`Unhandled content-encoding value "${contentEncoding}"`);
```
Comment (timdawborn) [inline src/http.ts:73]: Moved this common logic into `compression.ts`.

File: package.json [hunk 4]
```diff
@@ -51,7 +51,6 @@
   },
   "dependencies": {
     "assert-never": "^1.2.1",
-    "brotli": "^1.3.3",
```
Comment (timdawborn) [inline package.json:54]: This library has been broken on versions of node > 16. There's currently no tests in this codebase for anything brotli related 🙃 

Replacing this third-party library with the brotli implementation provided by the NodeJS builtin `zlib` module.

File: src/persistence.spec.ts [hunk 5]
```diff
@@ -221,7 +222,7 @@ describe("Persistence", () => {
         },
         body: {
           encoding: "base64",
-          data: "GxcAAI6UrMm1WkAERl0HoDFuCn3CIekc",
+          data: "GxcA+I+UrMm1WkAERl0HoDFuCn3CAZLOAQ==",
```
Comment (timdawborn) [inline src/persistence.spec.ts:225]: The brotli encoding default compression configuration differs between the old library and the new NodeJS builtin implementation. This is why hard-coded expected encoding here differs.

**Review by njday — DISMISSED** [review]
LGTM 👍 

Nice cleanups in this repo

**Review by njday — APPROVED** [review]

**Pull request merged**